python/cudf/cudf/core/udf/groupby_typing.py

```diff
@@ -0,0 +1,226 @@
+# Copyright (c) 2020-2023, NVIDIA CORPORATION.
+from typing import Any, Dict
+
+import numba
+from numba import cuda, types
+from numba.core.extending import (
+    make_attribute_wrapper,
+    models,
+    register_model,
+    type_callable,
+    typeof_impl,
+)
+from numba.core.typing import signature as nb_signature
+from numba.core.typing.templates import AbstractTemplate, AttributeTemplate
+from numba.cuda.cudadecl import registry as cuda_registry
+from numba.np import numpy_support
+
+# Disable occupancy warnings to avoid polluting output when there are few
+# groups.
+numba.config.CUDA_LOW_OCCUPANCY_WARNINGS = 0
+
+index_default_type = types.int64
```

Comment:
Is there a way to get this from an `Index` object instead? I am leery of encoding "defaults" in more than one place.

Reply:
Updated the this to take the pandas default index dtype. I'm thinking about a better way to do this the main barrier being the current import structure and circular import issues.